### ISSUE TITLE
linter not using eslintConfig from package.json

### ISSUE THREAD

**Issue by Reiner0** [AnomalyInnovations/serverless-bundle#9]
I know in the docs it references using .eslintrc.json to override the linter settings but I've tried that as well as my preferred way of having those settings specified in package.json, but it isn't working.

Here's the config from package.json:

```
"eslintConfig": {
    "extends": [
      "eslint:recommended",
      "google"
    ],
    "rules": {
      "no-console": 1,
      "max-len": [
        1,
        100,
        2
      ]
    },
    "parserOptions": {
      "ecmaVersion": 6,
      "ecmaFeatures": {
        "experimentalObjectRestSpread": true
      },
      "sourceType": "module"
    },
    "env": {
      "es6": true,
      "amd": true,
      "node": true
    },
    "overrides": [
      {
        "files": [
          "*test.js"
        ],
        "rules": {
          "no-undef": "off"
        }
      }
    ]
  }
```

and my devDeps:

```
"devDependencies": {
    "serverless-plugin-aws-alerts": "^1.4.0",
    "serverless-plugin-ifelse": "^1.0.4",
    "eslint-config-google": "^0.13.0",
    "serverless-bundle": "^1.2.2"
  }
```

From serverless.yml, I have this under "custom":

```
bundle:
    sourcemaps: true
    caching: true
    stats: false
    linting: true
```

When trying to run sls invoke local for a function, the bundler runs, but then the linter is throwing, despite the rules overrides in eslintConfig:

![image](https://user-images.githubusercontent.com/6852193/61891609-a0c15980-aecf-11e9-8096-023754abaa23.png)

**Comment by jayair** [comment]
I don't know if you can use a completely different config like `eslint-config-google`. I'll have to give that a try. But the overrides that you can apply are on top of the ones included in the plugin.

**Comment by Reiner0** [comment]
Cool!  I appreciate you looking into it.  If what I'm trying to do isn't support right now, I understand.  Maybe a nice-to-have for down the road.  :)

**Comment by Reiner0** [comment]
Something else I just noticed, this linting error only occurs when using `serverless invoke local`.  If I do `serverless webpack` or `yarn test` with scripts/test defined as `serverless-bundle test`, it doesn't occur.

**Comment by jayair** [comment]
Hmm that's interesting. I don't think there is a difference between the two commands but I'll have to check.

**Comment by Manzalber** [comment]
Hi @Reiner0 and @jayair the Readme is wrong it references that a _.eslintrc.json_ should be created in order to override ESLint config. The correct filename it's _.eslintrc_  from [eslint-config-strongloop](https://github.com/strongloop/eslint-config-strongloop) plugin info. Rename the file and it should take your config.

**Comment by Reiner0** [comment]
@Manzalber thanks!  I'll give it a try and report back here

**Comment by jayair** [comment]
Hmm that's weird because I have a test for this:

https://github.com/AnomalyInnovations/serverless-bundle/tree/master/tests/override-eslint

I'm using the `.eslintrc.json` there.

**Comment by wilcoschoneveld** [comment]
I'm confirming I also have this issue with `serverless offline`

**Comment by jayair** [comment]
@wilcoschoneveld Do you mean the eslint config is not getting picked up when you run `serverless offline`?

**Comment by wilcoschoneveld** [comment]
Yes, when I use `serverless invoke local` I have no issues. But when I run `serverless offline` the built-in eslint config seems to be used instead of my custom one.

**Comment by jayair** [comment]
Would it be possible to get a repo or simple set of instructions that I can test with?

**Comment by wilcoschoneveld** [comment]
Yes, I'm sorry I didn't provide one earlier. Here: https://github.com/wilcoschoneveld/serverless-bundle-example

Unfortunately I can't get the linting to work at all now, there is a `console.log` in the example handler but it doesn't get blocked by `serverless` commands. What is wrong in my setup? 

Thanks!

**Comment by jayair** [comment]
Thanks guys! I think I know what's going on. The issue is that this plugin uses Webpack caching. Meaning that when you make a config change, the cache needs to be cleared first.

``` bash
$ rm -rf node_modules/.cache
```

I added a note to the README with this - https://github.com/AnomalyInnovations/serverless-bundle#updating-options

@wilcoschoneveld I submitted a PR to your repo. Make sure you remove your `node_modules/` directory and `npm install` before testing.

**Comment by wilcoschoneveld** [comment]
Cool, thanks for looking into this.

I'm a bit skeptic about the proposed solution though. When working with teams this means that each member has to clear the cache when pulling via git if it contains any config changes. It might not happen often so it's easy to forget this step. What do you think?

**Comment by jayair** [comment]
Yeah I see what you mean. It doesn't happen often but it would be good if we could detect a change and clear the cache somehow.

**Comment by jayair** [comment]
I found that for the HardSourcePlugin we can set the cache to be set to be based on the environment or config - https://github.com/mzgoddard/hard-source-webpack-plugin#environmenthash. So there is a way to make it so that this cache is cleared when the config changes.

We also enable caching for babel-loader and eslint-loader. We need to check/test and see if they are cleared when the config changes.

**Comment by jayair** [comment]
I tested this a bit more. The caching issue is strictly for ESLint. We could disable the cache and bugs like this won't happen. https://github.com/AnomalyInnovations/serverless-bundle/blob/master/src/webpack.config.js#L65

Thoughts?

**Comment by wilcoschoneveld** [comment]
I like that idea :)

**Comment by jayair** [comment]
Yeah I'll test and see how much it affects performance. And maybe disable it if it's not too helpful.

**Comment by jayair** [comment]
The caching issue should be fixed in the next release. I'm closing this issue for now.
